Chore(docs): Add columns model property to docs (#1841)
TobikoData/sqlmesh: docs/concepts/models/overview.md

### clustered_by
- Clustered by is an optional property for engines such as Bigquery that support clustering.

### columns
- By default, SQLMesh [infers a model's column names and types](#conventions) from its SQL query. Disable that behavior by manually specifying all column names and data types in the model's `columns` property.
- **WARNING**: SQLMesh may exhibit unexpected behavior if the `columns` property includes columns not returned by the query, omits columns returned by the query, or specifies data types other than the ones returned by the query.

> - NOTE: Specifying column names and data types is required for [Python models](../models/python_models.md) that do not contain a SQL query.

Reviewer: ```suggestion
- NOTE: Specifying column names and data types is required for [Python models](../models/python_models.md) that return DataFrames.
```
